source/extensions/tracers/opentelemetry/tracer.cc

```diff
@@ -61,6 +61,30 @@ void Span::injectContext(Tracing::TraceContext& trace_context,
   trace_context.setByReferenceKey(kTraceParent, traceparent_header_value);
 }
 
+void Span::setTag(absl::string_view name, absl::string_view value) {
+  // The attribute key MUST be a non-null and non-empty string.
+  if (name.empty()) {
+    return;
+  }
+  // Attribute keys MUST be unique.
+  // If a value already exists for this key, overwrite it.
+  for (auto& key_value : *span_.mutable_attributes()) {
+    if (key_value.key() == name) {
+      key_value.mutable_value()->set_string_value(std::string{value});
+      return;
+    }
+  }
```

Comment:
This kind of N^2 behavior seems sub-optimal, though I guess there aren't usually that many tags. It might be worth it to optimize this later.